### ISSUE TITLE
why Gorm map property values not updated in  when update domain object

### ISSUE THREAD

**Issue by sadasivaiahgoli** [grails/gorm-mongodb#61]
I have the following domain objects

import org.bson.types.ObjectId

class Person {
    ObjectId id
    String firstName
    String lastName
    Map<String,Address> addresses
    static embedded = ['addresses']
}
import org.bson.types.ObjectId

class Address {
    ObjectId id
    String addressLine
    String addressLine2
    String street

}
I write the code to create the  person object with empty address
 Person person=new Person()
        person.firstName='SADASIVA'
        person.lastName='GOLI'
         person.save(flush:true)
It is saved in the collection as
{ 
    "_id" : ObjectId("5a21233d5cc6962b24afb675"), 
    "firstName" : "SADASIVA", 
    "lastName" : "GOLI", 
    "version" : NumberLong(5)
}
For updating the person Address I write the follwing logic
Person person=Person.findByFirstName('SADASIVA')
       
        Address paddress=new Address()
        paddress.addressLine='Desaipeta1'
        paddress.addressLine2='marrishettu'
        paddress.street='shanthinagar'
        Map<String,Address> map=new HashMap<String,Address>()
        map.put('perminentAddress',paddress)
        person.addresses=map
       person.save(flush:true)

The above code will not persist the address in the database while updating. How can we resolve this.
For Fresh new Object saving it is save the map (addresses property) also.
Thanks,
SADASIVAIAH GOLI

**Comment by graemerocher** [comment]
try call `person.markDirty("addresses")`

**Comment by sadasivaiahgoli** [comment]
I tried with markDirty but it is not working.

**Comment by graemerocher** [comment]
Attach an example that reproduces the issue please

**Comment by sadasivaiahgoli** [comment]
[my-app2.zip](https://github.com/grails/gorm-mongodb/files/1526920/my-app2.zip)


And give the url  localhost:8080/person/save . It will save the person only
localhost:8080/person/update . It will not save the address object as map in mongodb

**Comment by graemerocher** [comment]
Please upgrade to the latest versions of Grails, GORM and MongoDB which resolve the issue in the attached example. To do this set in `gradle.properties`:

```
grailsVersion=3.2.11
gormVersion=6.1.7.RELEASE
```

And in `build.gradle`:

```
compile "org.grails.plugins:mongodb:6.1.4"
```

**Comment by sadasivaiahgoli** [comment]
I even update the grails version and gorm version also . Map values are not updated in the object.
Iam using grails 3.3.10
gorm version :6.1.12 RELEASE
mongodb
compile 'org.grails.plugins:mongodb:6.1.7'
mongodb data base version: 3.4.21

**Comment by stesteau** [comment]
yeah, this should not be closed, indeed it doesn't work, the map never gets updated, the makeDirty does nothing, not idea why @graemerocher closed this issue, it doesn't even work in 7.3

At least provide some explanation, or don't support maps if it doesn't work
